lib/ex_json_schema/validator/pattern.ex

```diff
@@ -35,4 +36,13 @@ defmodule ExJsonSchema.Validator.Pattern do
   defp do_validate(_, _) do
     []
   end
+
+  @doc """
+  Converts ECMAScript style regexes to PCRE (for BEAM compatibility).
+  Currently supports `\\u` with 4 byte hex codes.
+  """
+  def convert_regex(r) do
+    r
+    |> String.replace(~r/\\u([A-F0-9]{4})/, "\\x\{\\g{1}\}")
```

Comment:
Nit: I prefer `String.replace(r, ~r/\\u([A-F0-9]{4})/, "\\x\{\\g{1}\}")` for single call pipelines like this, but it doesn't _actually_ matter. 🤷